test/src/unit-cppapi-string-dims.cc

```diff
@@ -109,4 +193,857 @@ TEST_CASE(
 
   if (vfs.is_dir(array_name))
     vfs.remove_dir(array_name);
-}
\ No newline at end of file
+}
+
+TEST_CASE(
+    "C++ API: Test default string dimensions",
+    "[cppapi][string-dim][default]") {
+  const std::string array_name = "cpp_unit_array";
+  Context ctx;
+
+  std::vector<char> dim1 = {'a', 'b', 'b', 'c', 'd', 'e', 'e', 'f'};
+  write_array_1(ctx, array_name, dim1, false);
+
+  std::vector<std::string> expected_dim1, collect_results_dim1;
+  std::vector<int32_t> expected_dim2, collect_results_dim2;
+  std::vector<std::string> expected_dim3, collect_results_dim3;
+  std::vector<int32_t> expected_a1_data, collect_results_a1;
+
+  auto perform_read = [&](int option = 5) {
+    collect_results_dim1.clear();
+    collect_results_dim2.clear();
+    collect_results_dim3.clear();
+    collect_results_a1.clear();
+    // Prepare a read query. Do not set a range for the last string dimension.
+    Array array_read(ctx, array_name, TILEDB_READ);
+    Query query_read(ctx, array_read, TILEDB_READ);
+    auto dim1_non_empty_domain = array_read.non_empty_domain_var(0);
+    auto dim2_non_empty_domain = array_read.non_empty_domain<int32_t>(1);
+    auto dim3_non_empty_domain = array_read.non_empty_domain_var(2);
+
+    // Each case section has the same .add_range possibilities,
+    // some active some inactive, to make visual comparison of what's
+    // different between them a bit easier, at least with some colorizing
+    // editors.
+    switch (option) {
```

Comment:
Really like all the options here! Any way we could use the catch SECTION macro?

Reply:
Yes, but...
Should a case fail, debugging the case as code is now requires adding only a single line perform_read(option_to_debug), with an eventual delete of that single line.
With SECTIONs, isolating one section would be more cumbersome to isolatea specific case, 
While 'tag's might be used for TEST_CASEs, to sub-select specific TEST-CASEs, SECTIONs do not appear to support 'tag's.

Reply:
Not that I tried but I think they might? https://github.com/catchorg/Catch2/blob/devel/docs/command-line.md#specify-the-section-to-run